features/poller/monitors/core/src/main/java/org/opennms/netmgt/poller/monitors/ActiveMQMonitor.java

```diff
@@ -0,0 +1,128 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author roskens
+ */
+@Distributable
+public class ActiveMQMonitor extends AbstractServiceMonitor {
+    private static final Logger LOG = LoggerFactory.getLogger(ActiveMQMonitor.class);
+
+    private static final int DEFAULT_RETRY = 3;
+    private static final int DEFAULT_TIMEOUT = 3000;
+    private static final String DEFAULT_BROKERURL = "vm://localhost?create=false&broker.persistent=false";
+
+    private static final String PARAMETER_BROKERURL = "brokerURL";
+    private static final String PARAMETER_USER = "user";
+    private static final String PARAMETER_PASSWORD = "password";
+    private static final String PARAMETER_USE_NODELABEL = "use-nodelabel";
+
+    @Override
+    public PollStatus poll(MonitoredService svc, Map<String, Object> parameters) {
+        PollStatus status = PollStatus.unknown("polling never attempted");
+
+        if (parameters == null) {
+                throw new NullPointerException("parameter cannot be null");
+        }
+
+        String brokerURL = ParameterMap.getKeyedString(parameters, PARAMETER_BROKERURL, DEFAULT_BROKERURL);
+        String userName  = ParameterMap.getKeyedString(parameters, PARAMETER_USER, null);
+        String password  = ParameterMap.getKeyedString(parameters, PARAMETER_PASSWORD, null);
+        Boolean useNodeLabel = ParameterMap.getKeyedBoolean(parameters, PARAMETER_USE_NODELABEL, false);
+        URI uri = null;
+        try {
+            uri = new URI(brokerURL);
+            LOG.debug("using brokerURL: {}", uri);
+            if (useNodeLabel) {
+                uri = new URI(uri.getScheme(), uri.getUserInfo(), svc.getNodeLabel(), uri.getPort(), uri.getPath(), uri.getQuery(), uri.getFragment());
+                LOG.debug("using updated brokerURL: {}", uri);
+            }
+        } catch (URISyntaxException e) {
+            throw new IllegalArgumentException("broker URL cannot be parsed");
+        }
+
+        ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory(uri);
+        if (userName != null && !userName.isEmpty()) {
+            connectionFactory.setUserName(userName);
+        }
+        if (password != null && !password.isEmpty()) {
+            connectionFactory.setPassword(password);
+        }
+
+        TimeoutTracker timeoutTracker = new TimeoutTracker(parameters, DEFAULT_RETRY, DEFAULT_TIMEOUT);
+        Connection connection = null;
+        Session session = null;
+
+        for(timeoutTracker.reset(); timeoutTracker.shouldRetry(); timeoutTracker.nextAttempt()) {
+            timeoutTracker.startAttempt();
+            status = PollStatus.unknown("polling never attempted");
+            try {
+                connection = connectionFactory.createConnection();
+                status = PollStatus.unresponsive("connected, no session");
+                session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+                connection.start();
```

Comment:
I think its enough to start the connection. From reviewing the source for ActiveMQ, it looks like when the connection factory creates the underlying transport, that initiates the actual socket to the broker. Then connection.start() ensures that the connection information has been sent to the broker.

As for the topic/queue, I think those could be configuration options for the monitor, or an extended version of this monitor.